pkg/reconciler/certificate/resources/service.go

```diff
@@ -25,18 +25,29 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/validation"
 )
 
 const portName = "http-challenge"
 
+func ServiceName(cert *v1alpha1.Certificate) string {
+	// Service names must be a DNS-1035 label. We try to use the
+	// Certificate name first if possible.
+	if name := kmeta.ChildName(cert.Name, ""); 0 == len(validation.IsDNS1035Label(name)) {
+		return name
+	}
+	// Fall back to a less readable name but guaranteed to be DNS-1035 label.
+	return "challenge-for-" + string(cert.GetUID())
```

Comment:
Are we guaranteed that this fits or should we use `kmeta.ChildName()` to be sure?

Ack the fixed length, but your tests aren't using a full uuid, so I wanted to double check

Reply:
Done now